tools/rsdl/abnf/rsdl.abnf

```diff
@@ -99,19 +122,199 @@ annotationValue      = %s"true"
+
+collectionRefValuedOperation = identifier          ; an operation that returns a collection of reference values
+
+castSegment = typeName
+
+keySegment = "{" keyProperty "}"
+
+keyProperty = identifier                           ; name of the key property
+
+singleNavigationProperty = identifier              ; a single valued navigation property
+
+nullableNavigationProperty = identifier            ; a nullable, single valued navigation property
+
+collectionNavigationProperty = identifier          ; a collection valued navigation property
+
+
+
+;----------------------------
+;  Path Capabilities
+;----------------------------
+
+
+singlePathCapability = ("GET" / "PUT" / "PATCH") [noOptions]
+
+singlePathCapabilities = "{" [singlePathCapability *( separator singlePathCapability)] "}"
+
+nullablePathCapability = singlePathCapability / "DELETE" noOptions
+
+nullablePathCapabilities = "{" [ nullablePathCapability *( separator nullablePathCapability )] "}"
+
+collectionPathCapability = "GET" [ collectionCapabilities ] / "POST" [noOptions]
+
+collectionPathCapabilities = "{" [ collectionPathCapability *( separator collectionPathCapability )] "}"
+
+singleRefPathCapability = ("GET" / "PATCH" / "PUT") [ OWS refCapabilities ]
+
+singleRefPathCapabilities = "{" [singleRefPathCapability *( separator singleRefPathCapability )] "}"
+
+nullableRefPathCapability = singleRefPathCapability / "DELETE" noOptions
+
+nullableRefPathCapabilities = "{" [nullableRefPathCapability *( separator nullableRefPathCapability )] "}"
+
+collectionRefPathCapability = "GET" [ OWS collectionRefCapabilities ] / "POST" [ OWS refCapabilities ]
+
+collectionRefPathCapabilities = "{" [ collectionRefPathCapability *( separator collectionRefPathCapability )] "}"
+
+
+
+;----------------------------
+;  Capability Elements
+;----------------------------
+
+
+collectionCapability = filterCapability / orderByCapability / "top" / "skip" / "count"
+
+collectionCapabilities =  "{" OWS [ collectionCapability *( separator collectionCapability ) OWS ] "}"
+
+collectionRefCapability = collectionCapability / refCapability
+
+collectionRefCapabilities = "{" OWS [ collectionRefCapability *( separator collectionRefCapability ) OWS ] "}"
+
+refCapability       = "expand" [ OWS "(" OWS [ expandProperty *( OWS "," OWS expandProperty OWS ) ] OWS ")" ]
+
+refCapabilities     =  "{" OWS [ refCapability OWS ] "}"
+
+expandProperty      = star /
+                      [ castSegment "/" ] collectionNavigationProperty [ OWS collectionRefCapabilities ] /
+                      [ castSegment "/" ] ( singleNavigationProperty / nullableNavigationProperty ) [ OWS refCapabilities ]
+
+filterCapability    = "filter" [ "(" [ OWS filterProperty *( "," OWS filterProperty OWS ) ] ")" ]
+
+filterProperty    = ( ( [ typeName "/" ] propertyName ) / allProperties) [ OWS filterOptions ]
+
+allProperties       =  star [ "/" typeName ]  ; all properties, optionally of a given type
```

Comment:
`*/this.Manager` meaning "all properties of the manager type" looks funny, and why would one need that?

If there is a use case, then switching the order and make it look like a type-cast would be more consistent with existing syntax: `this.Manager/*`

Reply:
Actually, */typeName was intended to mean all properties that were of a particular type -- for example, */String would be all string properties. So, for example,  "filter(*/String {stringComp}, */Integer {eq}" would mean you could filter all string properties by the string comparison operators and all integer properties using the eq comparison operator